### PR TITLE
add transcript for episode 034

### PR DESCRIPTION
**Pull request by ericseppanen**
Note: won't pass CI, as the episode page doesn't exist yet.

### CODE REVIEW

File: _transcripts/034-rust-1.52-1.53.md [hunk 1]
```diff
@@ -0,0 +1,1020 @@
...
+gives you a by-value one.
+
+__Ben__: OK. You're right, yes.
+
+__Jon__: The next one is, and this is I think a big one for me, is that in the
+2021 edition, closures are going to capture only the specific fields that are
+moved into them. So if you have a struct that has two fields `x` and `y`, and
+you only access, like, `struct.x` inside of a closure, and let's say you have a
+type `a`, that's an instance of that struct and you access like `a.y` inside the
+closure, but you don't access `a.x`. Then the closure will only move `a.y` into
```
Comment (ericseppanen) [inline _transcripts/034-rust-1.52-1.53.md:823]: `struct.x` is bad syntax, but I couldn't think of a better way to format this.

File: _transcripts/034-rust-1.52-1.53.md [hunk 2]
```diff
@@ -0,0 +1,1020 @@
...
+have your array `.into_iter().cloned()` to get things out by value. And, like,
+if you have, you know, if types are `Copy`, it's not going to be super expensive
+to get them out, but really you should be able to just do this and it's really
+cool that they found a way to make it happen. Do you want to talk about how they
+made it happen, Jon?
+
+Yeah, the— I mean, the hack is is sort of beautifully simple. The basic idea is
+that in the 2015 and 2018 editions, so basically in all the current editions, if
+you have an array and you call `into_iter` on it, then autoref will sort of take
+place. So autoref is this— we're going to call it a compiler feature or language
```
Comment (ericseppanen) [inline _transcripts/034-rust-1.52-1.53.md:387]: autoref? auto-ref? auto ref?

Comment (jonhoo) [reply _transcripts/034-rust-1.52-1.53.md:387]: I think autoref [is what the compiler uses](https://github.com/rust-lang/rust/blob/dddebf94bccddaa7b8836380c1d90b34553d79d0/compiler/rustc_infer/src/infer/error_reporting/mod.rs#L2420).

**Review by jonhoo — APPROVED** [review]

**Comment by ericseppanen** [comment]
Should be ready to merge.

**Pull request merged**